include/vast/ABI/Classify.hpp

```diff
@@ -232,6 +243,86 @@ namespace vast::abi
             return mlir::IntegerType::get( t.getContext(), final_size );
         }
 
+        static auto get_is_sse_type() {
+            return [](auto t) {
+                return is_scalar_float(t);
+            };
+        }
+
+        // TODO: In `clang` int and sse variants have separate implementations (there is
+        //       a lot of difference) but in the core they are doing very similar things.
+        //       It should be possible to make a shared helper that takes away a lot of
+        //       shared complexity.
+        static mlir_type sse_target_type_at_offset(
+            mlir_type t, std::size_t offset,
+            mlir_type root, std::size_t root_offset,
+            const auto &ctx
```

Comment:
```suggestion
            const auto &abi_ctx
```

Reply:
In general, it will improve readability if you add named type, e.g. `abi_ctx_t` and use that instead of auto if possible. Also use `abi_ctx` everywhere where applicable, as we have various contexts and one need to backtrack where these originates from.

Reply:
Fair enough, although this is not exactly `abi_ctx`, but I will come up with a good name for it.